apps/wallet/src/stacks/Explorer/CollectionStack/Collectible.tsx

```diff
@@ -1,24 +1,34 @@
-import { useEffect, useMemo } from 'react';
+import { useEffect, useMemo, useState } from 'react';
 import { Image, ScrollView, StyleSheet } from 'react-native';
 import { useRoute } from '@react-navigation/native';
 import { Button, Text, View } from '@walless/gui';
 import { showSendTokenModal } from 'modals/SendToken';
+import type { WrappedCollection } from 'utils/hooks';
 import { useNfts } from 'utils/hooks';
-import { navigationRef } from 'utils/navigation';
+import { navigateBack } from 'utils/navigation';
 
 export const CollectibleFeat = () => {
+	const [curCollection, setCurCollection] = useState<WrappedCollection>();
 	const { collections, collectibles } = useNfts();
 	const {
 		params: { id = '' },
 	} = useRoute() as never;
 
 	const curCollectible = useMemo(() => {
 		if (!id) return;
-		return collectibles.find((ele) => ele._id.includes(id));
+
+		const collectible = collectibles.find((ele) => ele._id.includes(id));
+		return collectible;
 	}, [collectibles]);
 
-	const curCollection = useMemo(() => {
-		return collections.find((ele) => ele._id === curCollectible?.collectionId);
+	useMemo(() => {
```

Comment:
useMemo being called/used as it is useEffect, use it this way cause confusion understanding the code.